src/main/scala/it/unibo/authsim/library/dsl/policy/builders/PolicyBuilder.scala

```diff
@@ -52,18 +54,35 @@ trait PolicyBuilder extends Builder[Policy]:
  *  PolicyBuilder is an implementation of policy builder
  */
 object PolicyBuilder:
-  def apply(): PolicyBuilder = new PolicyBuilderImpl();
-  def apply(policyName: String): PolicyBuilder = new PolicyBuilderImpl(policyName);
+  def apply(): PolicyBuilder = BasicPolicyBuilder();
+  def apply(policyName: String): PolicyBuilder = BasicPolicyBuilder(Some(policyName));
 
-  private class PolicyBuilderImpl(private val name: String = "") extends PolicyBuilder:
+  case class BasicPolicyBuilder(private val name: Option[String] = None) extends PolicyBuilder:

```

Comment:
This `case class` can remain private as it does not seem to flow outside the object (the `apply` return types are `PolicyBuilder`, so outside the user does not know what class there is under the hood).

```suggestion
  private case class BasicPolicyBuilder(private val name: Option[String] = None) extends PolicyBuilder:
```